lib/vsc/utils/fancylogger.py

```diff
@@ -562,6 +654,35 @@ def _logToSomething(handlerclass, handleropts, loggeroption, enable=True, name=N
     return handler
 
 
+def _screenLogFormatterFactory(colorize=Colorize.NEVER, stream=sys.stdout):
+    """
+    Return a log formatter class, with optional colorization features.
+
+    Second argument `colorize` controls whether the formatter
+    can use ANSI terminal escape sequences:
+
+    * ``Colorize.NEVER`` (default) forces use the plain `logging.Formatter` class;
+    * ``Colorize.ALWAYS`` forces use of the colorizing formatter;
+    * ``Colorize.AUTO`` selects the colorizing formatter depending on
+      whether `stream` is connected to a terminal.
+
+    Second argument `stream` is the stream to check in case `colorize`
+    is ``Colorize.AUTO``.
+    """
+    formatter = logging.Formatter  # default
+    if HAVE_COLOREDLOGS_MODULE:
+        if colorize == Colorize.AUTO:
+            # auto-detect
+            if humanfriendly.terminal.terminal_supports_colors(stream):
+                formatter = coloredlogs.ColoredFormatter
+        elif colorize == Colorize.ALWAYS:
+            formatter = coloredlogs.ColoredFormatter
+        else:
+            assert colorize == Colorize.NEVER, \
```

Comment:
There should be no performane penalty, since the `else` will never be reached in production? :)